rust/main/agents/scraper/src/db/message.rs

```diff
@@ -178,20 +179,20 @@ impl ScraperDb {
             })
             .collect_vec();
 
-        debug_assert!(!models.is_empty());
-        trace!(?models, "Writing delivered messages to database");
-
-        Insert::many(models)
-            .on_conflict(
-                OnConflict::columns([delivered_message::Column::MsgId])
-                    .update_columns([
-                        delivered_message::Column::TimeCreated,
-                        delivered_message::Column::DestinationTxId,
-                    ])
-                    .to_owned(),
-            )
-            .exec(&self.0)
-            .await?;
+        if !models.is_empty() {
```

Comment:
you can also just `return Ok(0)` if `models.is_empty()` to avoid the new indentation

Reply:
Fixed.